enumeratum-scalacheck/src/main/scala/enumeratum/CogenInstances.scala

```diff
@@ -0,0 +1,10 @@
+package enumeratum
+
+import org.scalacheck.Cogen
+
+trait CogenInstances {
+
+  implicit def cogenEnumEntry[EnumType <: EnumEntry]: Cogen[EnumType] =
```

Comment:
I'm putting this comment here, but it applies to all instances of `Cogen`.

I have found, when writing `enumeratum` instances for my various kantan libraries, that even if requiring an implicit `Enum` instance is not strictly necessary, it helps the compiler a *lot* when resolving implicits - without the constraint some code took entire minutes to compile, and it became instantaneous with the constraint.

I decided to add the constraint in my code because:
* it improves the experience so much
* it has no cost
* unless I'm mistaken, every sane `EnumEntry` subtype *will* always have an implicit `Enum` in the global implicit scope.

Reply:
That sounds like an interesting finding and if it improves the development experience, I think it's a great idea to add that constraint. 

That said, unless I'm missing something (completely possible!), it doesn't look like there is any constraint here that `EnumEntry` has an `Enum` instance (e.g. `EnumType <: EnumEntry : Enum`)?

Reply:
That's what Nicolas is talking about (adding `: Enum` constraint to `EnumType`).

In `Cogen` signature I didn't put it because it is not actually needed here (unlike for `Arbitrary` where I actually use the `Enum` instance to get the values), but from Nicolas' experience, adding it (although we won't use it) can make compilation faster.

The downside is that we won't be able to derive `Cogen` for `EnumEntry` which has no associated `Enum`, but that would be a weird situation anyway…

So if you're OK with it I can probably add this constraint.

Reply:
@nrinaudo @Sufflope Ugh, apologies, I didn't even realise that @nrinaudo was not the one who issued this PR and thought he was reviewing his own PR 😅 

@Sufflope I think adding the `: Enum` constraint would be reasonable. Asides from improving the dev experience (minutes -> instant sounds amazing, though I'm now more curious to understand why??), as you said, `Cogen` for `EnumEntry` with no `Enum` is kind of strange anyway.

So yea, if it isn't too much trouble can you please add the constraint (and perhaps in the code, leave a small link to this comment thread, so nobody feels the itch to take it wawy)? 

Thanks in advance :) and once again, sorry for my mistake.

Reply:
No worries, it's kind of rare for random people to ninja-review PRs like that. For context, @Sufflope and I are colleagues, that's how I knew to take a look.

The minutes -> instant thing was with more complicated types - basically `(A: Enum) => Either[E, B]`. My reasoning was that adding an `Enum` constraint reduces the search space drastically - most types don't have an `Enum` instance in scope, but most types *might* be a subtype of `EnumEntry`. If the compiler starts by applying the typeclass constraint before checking the subtyping one, it kind of makes sense. Maybe. Or maybe I've no idea what I'm talking about...